onionshare_gui/share_mode/threads.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+OnionShare | https://onionshare.org/
+
+Copyright (C) 2014-2018 Micah Lee <micah@micahflee.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from PyQt5 import QtCore
+
+
+class CompressThread(QtCore.QThread):
+    """
+    Compresses files to be shared
+    """
+    success = QtCore.pyqtSignal()
+    error = QtCore.pyqtSignal(str)
+
+    def __init__(self, mode):
+        super(CompressThread, self).__init__()
+        self.mode = mode
+        self.mode.common.log('CompressThread', '__init__')
+
+    # prepare files to share
+    def set_processed_size(self, x):
+        if self.mode._zip_progress_bar != None:
+            self.mode._zip_progress_bar.update_processed_size_signal.emit(x)
+
+    def run(self):
+        self.mode.common.log('CompressThread', 'run')
+
+        try:
+            if self.mode.web.set_file_info(self.mode.filenames, processed_size_callback=self.set_processed_size):
+                self.success.emit()
```

Comment:
This conditional is not succeeding for some reason, which is why `self.success.emit()` is not firing (which gives the experience of being 'blocked' at 'Compressing files'. It evaluates to false.

Reply:
Maybe because web.set_file_info() doesn't return anything? We didn't used to have it in an if/else condition. Instead we just ran it, it was already in a try/catch.  Removing the condition here, and forcing the self.success.emit(), of course works.. but you know better than me if that's OK. Anyway, this is where the bug is.

Reply:
Yes I was wrong that it was 'evaluating to false', I think the core of the issue is that `web.set_file_info()` needs to *return* something in order for this condition to pass..

Reply:
Oh yeah, I should have tested a bit more last night. Working on a fix now